driver/driver.go

```diff
@@ -89,11 +95,26 @@ type Driver interface {
 	Parse(ctx context.Context, src string, opts *ParseOptions) (nodes.Node, error)
 }
 
+// Version information for driver or the server.
+type Version struct {
+	Version string
+	Build   *time.Time
```

Comment:
Yep, I'm aware of this convention, but the driver manifest historically defined it as `*time.Time`. 

Although I would be glad to change it in both places if it's okay to break the manifest that way.

Manifest itself is used in a lot of places, but the build date is not, AFAIK.

Reply:
It appears that the Manifest is only for communication between the daemon and the drivers, in which case I think we "own" all the places it's used and wouldn't need to worry about breaking other clients—is that right?

From what I've seen, prior to this change we have only ever read manifests from files, and the decoder doesn't (I think) care whether the type is a pointer or not, since it uses reflection to populate the field. Now of course we're assembling them, so if it is not a huge pain to change both this would be a good time for it.

Reply:
We sometimes use the manifest for tooling, like in the documentation, or for the driver discovery, but those cases can be considered internal. Also, those cases don't use `Build` directly.

Tried to do this locally, it removes a lot of conversions, but the encoding breaks. TOML that we use for the manifest always encodes zero time as `0000-01-01T00:00:00Z` or as an empty string (when redefining `time.Time` with a custom text encoder).

This might not be a big issue if we redefine the encoder and set the value to an empty string, but it makes it a bit harder to work with the time value since it requires an additional cast to `time.Time`.

I may try to fix upstream as an alternative so TOML works properly for `time.Time` :grin:

Reply:
Ugh. It probably is a bug that the TOML package's `omitempty` metatag doesn't work for the zero value, but I agree it is probably not worth delaying our work to fix that.  

It's probably worth adding a comment, though, because this is the kind of thing I'd clean up if I stumbled on it out of context. 😄 

```go
   Build *time.Time // the timestamp when the driver was built

   // Note that we use a pointer here so that the TOML encoding will work 
   // correctly even when we don't know the build time.
}
```

Actually, that makes me wonder: How often do we actually _encode_ manifest files for re-use, vs. just decoding them from ones that the driver maintainer has written?

Reply:
I think we only encode once when initializing a new driver. After that, we don't touch it to not wipe developer comments in that file.

There is a second case when we re-write manifest to build the final image, but in that case, we always include the build time.

Let me check if we actually use the encoder when initializing the driver. It may be writing that file from Go templates instead.

Reply:
Confirmed, we aren't using the encoder in any places except the build. Should be safe to remove the pointer then.

Reply:
Done